src/track/track.cpp

```diff
@@ -830,9 +830,14 @@ bool Track::isDirty() {
 
 void Track::markForMetadataExport() {
     QMutexLocker lock(&m_qMutex);
-    if (compareAndSet(&m_bMarkedForMetadataExport, true)) {
-        markDirtyAndUnlock(&lock);
-    }
+    m_bMarkedForMetadataExport = true;
```

Comment:
I think we should also unset this flag after export. It does not cost much, and we do not need to worry about it if we discuss the changes in the concurrent cache PR. 
A similar issue would be to re-allow signals from the track.